OpenSim/Simulation/Model/Appearance.h

```diff
@@ -33,105 +33,160 @@ namespace OpenSim {
+};  // END of class CurveAppearance
 
 //=============================================================================
-class OSIMSIMULATION_API LineAppearance : public BaseAppearance {
-    OpenSim_DECLARE_CONCRETE_OBJECT(LineAppearance, Object);
+//=============================================================================
+/**
+ * A class that holds the Appearance properties of an object displayed 
+ * in the OpenSim Visualizer.
+ * 
+ * Appearance contains properties that apply to all geometry.
+ * Geometry that have a surface so that it can be textured will utilize
+ * SurfaceAppearance, while schematic line drawings (e.g. Arrows, Frames) 
+ * can utilize CurveAppearance which offers thickness. 
+ *
+ * @author Ayman Habib
+ */
+class OSIMSIMULATION_API Appearance : public Object {
+    OpenSim_DECLARE_CONCRETE_OBJECT(Appearance, Object);
 public:
-    //==============================================================================
+
+    //==========================================================================
     // PROPERTIES
-    //==============================================================================
-    OpenSim_DECLARE_PROPERTY(thickness, double,
-        "The thickness used to visualize a LineGeometry. ");
+    //==========================================================================
+    OpenSim_DECLARE_PROPERTY(visible, bool, 
+        "Flag indicating whether the associated Geometry is visible or hidden.")
+    OpenSim_DECLARE_PROPERTY(opacity, double,
+            "The opacity (0-1) used to display the object. ");
+    OpenSim_DECLARE_PROPERTY(color, SimTK::Vec3,
+        "The color (rgb) used to display the object. ");
+
+    OpenSim_DECLARE_PROPERTY(surface_appearance, SurfaceAppearance,
+        "Visuals applied to surfaces associated with this Appearance. ");
+    OpenSim_DECLARE_PROPERTY(curve_appearance, CurveAppearance,
+        "Visuals applied to curves or line drawings associated with this Appearance. ");
+
 
     //--------------------------------------------------------------------------
     // CONSTRUCTION
     //--------------------------------------------------------------------------
 public:
-    LineAppearance() {
+    Appearance() {
         constructProperties();
     }
-    virtual ~LineAppearance() {};
+    virtual ~Appearance() {};
+
+    VisualRepresentation get_representation() const { 
+        return (VisualRepresentation)get_surface_appearance().get_representation(); }
+
+    void set_representation(VisualRepresentation& rep) { 
+        upd_surface_appearance().set_representation(rep); }
+
+protected:
+    /** Updating XML formating to latest revision */
+    void updateFromXMLNode(SimTK::Xml::Element& aNode, int versionNumber) override {
+        int documentVersion = versionNumber;
+        if (documentVersion < XMLDocument::getLatestVersion()) {
+            if (documentVersion < 30505) {
+                // if representation = 0 visible = false, else true
```

Comment:
if A <i>then</i> B. Else .. 
Where is Appearance added to XML in old formats that do not have Appearance?